### PR TITLE
Smartport telem : optimize conditional build tags, airmode support

### PR DESCRIPTION
**Pull request by 3djc**
Minor changes on smart telemetry, should not have user impact

### CODE REVIEW

File: src/main/telemetry/smartport.c [hunk 1]
```diff
@@ -379,8 +377,10 @@ void handleSmartPortTelemetry(void)
                 }
                 break;
             case FSSP_DATAID_HEADING    :
-                smartPortSendPackage(id, attitude.values.yaw * 10); // given in 10*deg, requested in 10000 = 100 deg
-                smartPortHasRequest = 0;
+		if (sensors(SENSOR_MAG)) {
```
Comment (ledvinap) [inline src/main/telemetry/smartport.c]: Why? heading is available even without magnetometer (with usable short-term relative accuracy)

**Comment by 3djc** [comment]
Did not realize this. What is the reference then ? Power up position ? (will update doc accordingly)

**Comment by ledvinap** [comment]
@3djc : It's probably undefined/dependent on power-on attitude

**Comment by 3djc** [comment]
I see, so you could only use for very term 'i'm yawing left or right' kinda thing. Is it really worth transmiting through telemetry in this case ? (I can see a value to sending hdg with mag, unsure without given what you describe)

**Comment by ledvinap** [comment]
OpenTX can record telemetry on SD card .. at least for plane even relative heading is interresting

**Comment by 3djc** [comment]
ok, I can see a usage scenario then :)

**Comment by hydra** [comment]
this pr is too unfocussed to merge, it's also out of date.  I'll close this due to lack of interest and time.

**Pull request closed without merging**